### PR TITLE
Experiment with swagger nullable string

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the
2. Login as a
3.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/swagger/nullable/string.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	if err := json.Unmarshal(data, &s.Value); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// String always validates ok
+// this is called from the go swagger generated code
+func (s *String) Validate(formats strfmt.Registry) error {
```
Comment (eamahanna) [inline pkg/swagger/nullable/string.go:60]: Why are the validate and context validate functions needed?

Comment (ahobson) [reply pkg/swagger/nullable/string.go:60]: It's called from [the generated code](https://github.com/transcom/mymove/pull/7881/files#diff-0d162228bcafba3260f9b0f5a4773ceca7c8ed219bfe3db2a6fd40ad4476c8f3R268)

File: swagger-def/definitions/NullableString.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+type: string
+x-go-type:
+  import:
+    package: github.com/transcom/mymove/pkg/swagger/nullable
```
Comment (jacquelineIO) [inline swagger-def/definitions/NullableString.yaml:4]: @ahobson so are we creating our own types here and not re-using this other one?
```
go get gopkg.in/romanyx/nullable.v1
```
cc: @reggieriser @donaldthai @rogeruiz

Comment (ahobson) [reply swagger-def/definitions/NullableString.yaml:4]: At one point I think Emily was working on this and so I didn't try to finish/polish this draft PR. Maybe that work didn't finish and so I need to get this work ready to merge?

Comment (jacquelineIO) [reply swagger-def/definitions/NullableString.yaml:4]: @ahobson if you were going to do that, that might be good. I was about to recreate this PR. I guess no one knew you planned on finishing it. 

But my other question is, why is the `Present` field missing.

i.e.,

```
// String represents a string that may be null or not
// present in json at all.
type String struct {
	Present bool // Present is true if key is present in json
	Valid   bool // Valid is true if value is not null and valid string
	Value   string
}
```

In both of the mentioned examples of what we were trying to do, they use the `Present` field. But this PR doesn't have it. Was that on purpose?

Comment (jacquelineIO) [reply swagger-def/definitions/NullableString.yaml:4]: moved convo to slack https://ustcdp3.slack.com/archives/CP6PTUPQF/p1643305125135400

**Comment by ahobson** [comment]
This was adopted elsewhere

**Pull request closed without merging**